### PR TITLE
Cqw 159 menu changes

### PR DESCRIPTION
**Pull request by beatadelura**
## TO DO:
* [X] Change styles for all resolutions
* [X] Remove third state (active). Remove dot svg.
* [X] Disable scroll page when menu is open.
* [X] Allow to scroll menu when window height is smaller than menu content
* [X] Menu items shouldn't stretch via window heigh
* [X] Code refactor

---

https://codequest.atlassian.net/browse/CQW-159

---

Review: 
* [ ] @pjar 
* [x] @MagdaMalinowska 
* [ ] @rpietraszko 
* [ ] @kruszczynski 
* [ ] @AdamMankowski 
* [ ] @irene50288

### CODE REVIEW

File: source/stylesheets/layouts/_l-not-found-page.scss [hunk 1]
```diff
@@ -9,7 +9,7 @@
     }
   }
 
-  .three-states-button {
+  .m-burger-button {
     color: $martinique-dark;
```
Comment (pjar) [inline source/stylesheets/layouts/_l-not-found-page.scss]: Please move it to burger module file

File: source/stylesheets/base/_main.scss [hunk 2]
```diff
@@ -2,13 +2,13 @@
 @include bcg-images;
 
 .landing-page {
-  .three-states-button {
+  .m-burger-button {
     color: $martinique-dark;
```
Comment (pjar) [inline source/stylesheets/base/_main.scss]: Please move it to burger module file

File: source/stylesheets/base/_main.scss [hunk 3]
```diff
@@ -2,13 +2,13 @@
...
   }
 }
 
 .website-page {
-  .three-states-button {
+  .m-burger-button {
```
Comment (pjar) [inline source/stylesheets/base/_main.scss]: Please move it to burger module file

File: source/stylesheets/modules/_m-main-menu-navigation.scss [hunk 4]
```diff
@@ -7,7 +7,6 @@
 .main-menu-list {
   @include justify-content(space-between);
   @include flex-flow(column, nowrap);
-  height: 55vh;
```
Comment (pjar) [inline source/stylesheets/modules/_m-main-menu-navigation.scss]: Without this height, flex styling doesn't make sense anymore.

File: source/stylesheets/modules/_m-main-menu.scss [hunk 5]
```diff
@@ -121,28 +122,19 @@
...
+@media #{$breakpoint-desktop-large} {
+  .main-menu {
+    @include position(0, -80vw, 0, 136vw);
   }
+}
 
+@media #{$breakpoint-desktop-huge} {
   .main-menu {
-    @include position(0, -33vw, 0, 100vw);
+    @include position(0, -80vw, 0, 145vw);
```
Comment (pjar) [inline source/stylesheets/modules/_m-main-menu.scss]: 145vw here means that content will be placed 45vw from the right side of the screen - we probably don't need that :) 

It'll be addressed in separate ticket though.

File: source/stylesheets/layouts/_l-not-found-page.scss (hunk 1 above)
Comment (rpietraszko) [inline source/stylesheets/layouts/_l-not-found-page.scss]: Actually it was my mistake to put it here - I should have just used `color-martinique-dark` class instead (same for `o-shaped-submarine` few lines higher). I can make a fix after this PR is merged

File: source/stylesheets/base/_devices.scss [hunk 6]
```diff
@@ -13,4 +13,5 @@ $breakpoint-tablet: 'only screen and (min-width: 768px)';
 $breakpoint-desktop: 'only screen and (min-width: 997px)';
 $breakpoint-desktop-large: 'only screen and (min-width: 1248px)';
 $breakpoint-desktop-larger: 'only screen and (min-width: 1282px)';
+$breakpoint-desktop-huge: 'only screen and (min-width: 1400px)';
```
Comment (MagdaMalinowska) [inline source/stylesheets/base/_devices.scss]: large/larger/huge :D

File: source/stylesheets/layouts/_l-not-found-page.scss (hunk 1 above)
Comment (beatadelura) [inline source/stylesheets/layouts/_l-not-found-page.scss]: Ok, I moved it to `m-burger-button` module.

File: source/stylesheets/modules/_m-main-menu.scss (hunk 5 above)
Comment (beatadelura) [inline source/stylesheets/modules/_m-main-menu.scss]: Fixed.

File: source/stylesheets/modules/_m-main-menu-navigation.scss (hunk 4 above)
Comment (beatadelura) [inline source/stylesheets/modules/_m-main-menu-navigation.scss]: @pjar There are few things to do in menu yet. I did a task in jira https://codequest.atlassian.net/browse/CQW-184 What do you think to add this issue to as an AC here?

File: source/stylesheets/base/_devices.scss (hunk 6 above)
Comment (beatadelura) [inline source/stylesheets/base/_devices.scss]: Not needed now :) Removed.

File: source/stylesheets/modules/_m-main-menu-navigation.scss (hunk 4 above)
Comment (pjar) [inline source/stylesheets/modules/_m-main-menu-navigation.scss]: Let's just leave like it is for now.

File: source/stylesheets/modules/_m-main-menu.scss [hunk 7]
```diff
@@ -7,27 +7,27 @@
...
   100% {
-    @include transform(translate3d(-44vw, 0, 0));
+    @include transform(translate3d(-600px, 0, 0));
   }
 }
 
 @include keyframes(mainMenuBounceOut) {
   0% {
-    @include transform(translate3d(-30vw, 0, 0));
+    @include transform(translate3d(-32vw, 0, 0));
```
Comment (pjar) [inline source/stylesheets/modules/_m-main-menu.scss]: Why bounceOut is not using px? You might have the same problem as originally with vw in other places

Comment (beatadelura) [inline source/stylesheets/modules/_m-main-menu.scss]: :+1:

File: source/stylesheets/modules/_m-burger-button.scss [hunk 8]
```diff
@@ -1,50 +1,80 @@
...
+    }
+
+    .mobile-burger-btn {
+      display: none;
+    }
+  }
+}
+
+@media #{$breakpoint-tablet} {
+  .no-touchevents {
```
Comment (pjar) [inline source/stylesheets/modules/_m-burger-button.scss]: Just .no-touchevents is enough - we don't need media query for hovers

File: source/stylesheets/base/_main.scss [hunk 9]
```diff
@@ -48,6 +35,12 @@
   display: none;
 }
 
+.disable-scroll {
```
Comment (kruszczynski) [inline source/stylesheets/base/_main.scss]: Isn't this a problem on mobile when the menu is taller then the viewport?

File: source/stylesheets/modules/_m-burger-button.scss [hunk 10]
```diff
@@ -1,50 +1,78 @@
...
+  .desktop-burger-btn {
     @include size(100%, 100%);
     display: none;
   }
 
-  &:active {
+  .hover-arrow-btn {
+    @include size(100%, 100%);
+    display: none;
+  }
```
Comment (kruszczynski) [inline source/stylesheets/modules/_m-burger-button.scss]: That's brutal duplication

File: source/stylesheets/modules/_m-burger-button.scss [hunk 11]
```diff
@@ -1,50 +1,78 @@
...
-      .clicked-state,
-      .normal-state-normal {
+      .desktop-burger-btn {
         display: none;
       }
 
-      .hover-state {
-        display: inline-block;
+      .hover-arrow-btn {
+        display: none;
```
Comment (kruszczynski) [inline source/stylesheets/modules/_m-burger-button.scss]: This too

File: source/stylesheets/base/_main.scss (hunk 9 above)
Comment (beatadelura) [inline source/stylesheets/base/_main.scss]: I didn't notice any problem. We are adding this to html and body when menu is open. If menu is taller than vieport user can scroll menu (not the body).

File: source/stylesheets/modules/_m-burger-button.scss (hunk 10 above)
Comment (beatadelura) [inline source/stylesheets/modules/_m-burger-button.scss]: Fixed here: https://github.com/codequest-eu/website/pull/159

File: source/stylesheets/modules/_m-burger-button.scss (hunk 11 above)
Comment (beatadelura) [inline source/stylesheets/modules/_m-burger-button.scss]: Fixed here: https://github.com/codequest-eu/website/pull/159

**Pull request merged**